x/ibchooks/client/cli/cli_test.go

```diff
@@ -0,0 +1,181 @@
+package cli_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	cmtcli "github.com/cometbft/cometbft/libs/cli"
+
+	sdkmath "cosmossdk.io/math"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	clitestutil "github.com/cosmos/cosmos-sdk/testutil/cli"
+	"github.com/cosmos/cosmos-sdk/testutil/network"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	govv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+
+	"github.com/provenance-io/provenance/internal/antewrapper"
+	"github.com/provenance-io/provenance/internal/pioconfig"
+	"github.com/provenance-io/provenance/testutil"
+	testcli "github.com/provenance-io/provenance/testutil/cli"
+	ibchookscli "github.com/provenance-io/provenance/x/ibchooks/client/cli"
+	ibchookstypes "github.com/provenance-io/provenance/x/ibchooks/types"
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	cfg              network.Config
+	testnet          *network.Network
+	keyring          keyring.Keyring
+	keyringDir       string
+	accountAddr      sdk.AccAddress
+	accountKey       *secp256k1.PrivKey
+	accountAddresses []sdk.AccAddress
+}
+
+func TestIntegrationTestSuite(t *testing.T) {
+	suite.Run(t, new(IntegrationTestSuite))
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up integration test suite")
+	pioconfig.SetProvenanceConfig("", 0)
+	govv1.DefaultMinDepositRatio = sdkmath.LegacyZeroDec()
+	s.accountKey = secp256k1.GenPrivKeyFromSecret([]byte("acc2"))
+	addr, err := sdk.AccAddressFromHexUnsafe(s.accountKey.PubKey().Address().String())
+	s.Require().NoError(err)
+	s.accountAddr = addr
+
+	s.cfg = testutil.DefaultTestNetworkConfig()
+	genesisState := s.cfg.GenesisState
+
+	s.cfg.NumValidators = 1
+	s.generateAccountsWithKeyrings(2)
+
+	var genBalances []banktypes.Balance
+	for i := range s.accountAddresses {
+		genBalances = append(genBalances, banktypes.Balance{Address: s.accountAddresses[i].String(), Coins: sdk.NewCoins(
+			sdk.NewInt64Coin("nhash", 100_000_000), sdk.NewInt64Coin(s.cfg.BondDenom, 100_000_000),
+		).Sort()})
+	}
+	var bankGenState banktypes.GenesisState
+	bankGenState.Params = banktypes.DefaultParams()
+	bankGenState.Balances = genBalances
+	bankDataBz, err := s.cfg.Codec.MarshalJSON(&bankGenState)
+	s.Require().NoError(err, "should be able to marshal bank genesis state when setting up suite")
+	genesisState[banktypes.ModuleName] = bankDataBz
```

Comment:
Please use `testutil.MutateGenesisState` for this stuff.

Reply:
Oh yes, so much cleaner!  Copy and pasted this setup from somewhere else...  We should probably make an issue (`nice-to-have`) to clean this up to all use the `testutil.MutateGenesisState`.